src/Question.js

```diff
@@ -46,20 +46,26 @@ class Question extends React.Component {
         this.props.updateQuestionsState(this.state);
     };
 
-    
+    deleteQuestion = () => {
+        this.props.deleteQuestion(this.props.index);
+    }
+
+    deleteIncorrect = (event) => {
+        let incorrect = this.state.incorrect
+        incorrect.splice(event.target.attributes.index, 1)
+        this.setState({incorrect: incorrect});
+    }
 
     render() {
-        if (this.state.incorrect.length === 0) {
-            this.addIncorrect();
-        };
         let incorrectResponses = this.state.incorrect.map((val, index) => {
             return (
-                <Row fluid="md">
-                    <Form.Group key={index}>
+                <Row key={Math.random()} fluid="md">
```

Comment:
This isn't necessary. The only components that need a `key` (for 99.99% of cases) are what's immediately inside the `map()`. Everything inside those keyed components will automatically rerender or not based on changes to their props.